addon/components/ember-tooltip-base.js

```diff
@@ -292,14 +292,12 @@ export default Component.extend({
 
     this._addEventListener('keydown', (keyEvent) => {
 
-      keyEvent.stopImmediatePropagation(); /* So this callback only fires once per keydown */
-
       if (keyEvent.which === 27) {
-        this.hide();
-
-        keyEvent.preventDefault();
-
-        return false;
+        if (this.hide()) {
```

Comment:
`this.hide()` has async side-effects, so I don't think it's really suited to returning a value and being used in this way. Checking `this.get('isShown')` would be better, I think.

Reply:
Updated!